client/relay-chain-interface/src/lib.rs

```diff
@@ -0,0 +1,441 @@
+	}
+
+	fn import_notification_stream(&self) -> sc_client_api::ImportNotifications<Block> {
+		(**self).import_notification_stream()
+	}
+
+	fn finality_notification_stream(&self) -> sc_client_api::FinalityNotifications<Block> {
+		(**self).finality_notification_stream()
+	}
+
+	fn storage_changes_notification_stream(
+		&self,
+		filter_keys: Option<&[sc_client_api::StorageKey]>,
+		child_filter_keys: Option<
+			&[(sc_client_api::StorageKey, Option<Vec<sc_client_api::StorageKey>>)],
+		>,
+	) -> sc_client_api::blockchain::Result<sc_client_api::StorageEventStream<Block::Hash>> {
+		(**self).storage_changes_notification_stream(filter_keys, child_filter_keys)
+	}
+
+	fn best_block_hash(&self) -> PHash {
+		(**self).best_block_hash()
+	}
+
+	fn block_status(&self, block_id: BlockId) -> Result<BlockStatus, sp_blockchain::Error> {
+		(**self).block_status(block_id)
+	}
+
+	fn get_import_lock(&self) -> &RwLock<()> {
+		(**self).get_import_lock()
+	}
+
+	fn get_state_at(
+		&self,
+		block_id: BlockId,
+	) -> Result<<FullBackend as sc_client_api::Backend<PBlock>>::State, sp_blockchain::Error> {
+		(**self).get_state_at(block_id)
+	}
+
+	fn is_major_syncing(&self) -> bool {
+		(**self).is_major_syncing()
+	}
+
+	fn slot_duration(&self) -> Result<Duration, sp_blockchain::Error> {
+		(**self).slot_duration()
+	}
+
+	fn overseer_handle(&self) -> Option<Handle> {
+		(**self).overseer_handle()
+	}
+}
+
+/// Build the Polkadot full node using the given `config`.
+#[sc_tracing::logging::prefix_logs_with("Relaychain")]
+pub fn build_polkadot_full_node(
+	config: Configuration,
+	telemetry_worker_handle: Option<TelemetryWorkerHandle>,
+) -> Result<(NewFull<polkadot_client::Client>, CollatorPair), polkadot_service::Error> {
+	let is_light = matches!(config.role, Role::Light);
+	if is_light {
+		Err(polkadot_service::Error::Sub("Light client not supported.".into()))
+	} else {
+		let collator_key = CollatorPair::generate().0;
+
+		let relay_chain_full_node = polkadot_service::build_full(
+			config,
+			polkadot_service::IsCollator::Yes(collator_key.clone()),
+			None,
+			true,
+			None,
+			telemetry_worker_handle,
+			polkadot_service::RealOverseerGen,
+		)?;
+
+		Ok((relay_chain_full_node, collator_key))
+	}
+}
+
+/// Builds a relay chain interface by constructing a full relay chain node
+pub fn build_relay_chain_interface(
+	polkadot_config: Configuration,
+	telemetry_worker_handle: Option<TelemetryWorkerHandle>,
+	task_manager: &mut TaskManager,
+) -> Result<
+	(Arc<(dyn RelayChainInterface<PBlock> + Send + Sync + 'static)>, CollatorPair),
+	polkadot_service::Error,
+> {
+	let (full_node, collator_key) =
+		build_polkadot_full_node(polkadot_config, telemetry_worker_handle).map_err(
+			|e| match e {
+				polkadot_service::Error::Sub(x) => x,
+				s => format!("{}", s).into(),
+			},
+		)?;
+
+	let sync_oracle: Box<dyn SyncOracle + Send + Sync> = Box::new(full_node.network.clone());
+	let network = Arc::new(Mutex::new(sync_oracle));
```

Comment:
Sync oracle changes its name to network?

We should probably just call it `sync_oracle`